src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -149,26 +170,46 @@ public boolean requiresReference() {
     @Override
     public void onStartup() {
 
-        for (final String variantPath : variantFilePaths) {
+        List<VCFHeader> headers = new ArrayList<>(variantFilePaths.size());
 
-            final String variantURI = IOUtils.getPath(variantPath).toAbsolutePath().toUri().toString();
+        for (String variantFilePath : variantFilePaths) {
+
+            final String variantURI = IOUtils.getPath(variantFilePath).toAbsolutePath().toUri().toString();
             final Function<SeekableByteChannel, SeekableByteChannel> cloudWrapper = (cloudPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudPrefetchBuffer, is) : Function.identity());
             final Function<SeekableByteChannel, SeekableByteChannel> cloudIndexWrapper = (cloudIndexPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudIndexPrefetchBuffer, is) : Function.identity());
             final AbstractFeatureReader<VariantContext, LineIterator> reader = AbstractFeatureReader.getFeatureReader(variantURI, null, new VCFCodec(), true, cloudWrapper, cloudIndexWrapper);
 
+            VCFHeader header = (VCFHeader)reader.getHeader();
+
             // A GVCF file must contain only one sample, throw an exception otherwise
-            if (((VCFHeader)reader.getHeader()).getNGenotypeSamples() != 1) {
-                throw new UserException("Input GVCF: " + variantPath + " should contain data for one sample");
+            if (header.getNGenotypeSamples() != 1) {
+                throw new UserException("Input GVCF: " + variantFilePath + " should contain data for one sample");
+            }
+            String sampleName = header.getGenotypeSamples().get(0);
+            headers.add(header);
+            sampleNames.add(sampleName);
+            sampleNameToFileName.put(sampleName, variantFilePath);
```

Comment:
Check the return value of this `put()` call -- if the key was already present (indicated by a non-null return value), throw a `UserException` saying that there were duplicate samples.